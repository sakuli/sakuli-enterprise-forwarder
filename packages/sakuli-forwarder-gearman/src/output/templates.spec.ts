import { TemplateEnvironment } from "@sakuli/forwarder-commons";
import { promises as fs, readdirSync, readFileSync } from 'fs';
import { join } from "path";
import { TestExecutionContext, TestContextEntity, TestCaseContext, TestContextEntityKind } from "@sakuli/core/dist";
import { SimpleLogger } from "@sakuli/commons/dist";
import { GearmanForwarderProperties } from "../gearman-forwarder-properties.class";
import { create } from "domain";


describe('templates', () => {
    const templateDir = join(__dirname, '..', '..', 'templates');
    let templateEnvironment: TemplateEnvironment;
    let ctx: TestExecutionContext;

    const createEntity = (kind: TestContextEntityKind, data: Partial<TestContextEntity>, children: TestContextEntity[] = []): TestContextEntity => {
        return Object.assign(data, new (class extends TestContextEntity {
            kind = kind
            getChildren() {
                return children;
            }
        }));
    }

    const entities = {
        TestCase_OK: createEntity('case', {
            warningTime: 20,
            criticalTime: 30,
            startDate: new Date(1970, 1, 1, 10, 30, 0),
            endDate: new Date(1970, 1, 1, 10, 30, 14, 20),
            id: 'case1'
        }, [
                createEntity('step', {
                    id: 'Test_Sahi_landing_page',
                    warningTime: 5,
                    startDate: new Date(1970, 1, 1, 10, 30, 0),
                    endDate: new Date(1970, 1, 1, 10, 30, 1, 160)
                }),
                createEntity('step', {
                    id: 'Calculation',
                    warningTime: 10,
                    startDate: new Date(10970, 1, 1, 10, 30, 0, 10),
                    endDate: new Date(1970, 1, 1, 10, 30, 7, 290)
                }),
                createEntity('step', {
                    id: 'Editor',
                    warningTime: 10,
                    startDate: new Date(1970, 1, 1, 10, 30, 0, 20),
                    endDate: new Date(1970, 1, 1, 10, 30, 1, 500)
                })
            ]),
        TestCase_WARNING_IN_STEP: createEntity('case', {
            warningTime: 20,
            criticalTime: 30,
            startDate: new Date(1970, 1, 1, 10, 31, 20),
            endDate: new Date(1970, 1, 1, 10, 31, 33, 430),
            id: 'case2',
        }, [
                createEntity('step', {
                    id: 'Test_Sahi_landing_page_(case2)',
                    warningTime: 5,
                    startDate: new Date(1970, 1, 1, 10, 31, 0, 10),
                    endDate: new Date(1970, 1, 1, 10, 31, 0, 930),
                }),
                createEntity('step', {
                    id: 'Calculation_(case2)',
                    warningTime: 1,
                    startDate: new Date(1970, 1, 1, 10, 31, 0, 20),
                    endDate: new Date(1970, 1, 1, 10, 31, 7, 20)
                }),
                createEntity('step', {
                    id: 'Editor_(case2)',
                    warningTime: 10,
                    startDate: new Date(1970, 1, 1, 10, 31, 0, 30),
                    endDate: new Date(1970, 1, 1, 10, 31, 1, 420)
                })
            ])
    }

    beforeEach(() => {
        templateEnvironment = new TemplateEnvironment(templateDir);
        ctx = new TestExecutionContext(new SimpleLogger);

    });

    describe.each(Object.entries(entities))('%s for main.twig', (key: string, testDataEntity: TestContextEntity) => {

        let properties: Partial<GearmanForwarderProperties>;
        let tpl: any;
        let expected: string;
        beforeEach(() => {
            properties = {
                nagiosHost: 'my.nagios.host'
            }
            tpl = templateEnvironment.getTemplate('main.twig');
            expected = readFileSync(join(__dirname, '__snapshots__', key + '.txt')).toString();
        })

        it('should match static file', () => {
            expect(tpl.render({
                gearman: properties,
                testDataEntity
            })).toEqual(expected)
        })

    })

})