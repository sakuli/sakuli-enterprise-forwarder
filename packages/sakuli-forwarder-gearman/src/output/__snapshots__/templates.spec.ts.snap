// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates main.twig should match snapshot for Suite 1`] = `
"type=passive

    host_name=
            changeme
    

    start_time=1556540579.709

    finish_time=-1

    return_code=

                service_description=\${testsuite.id}

        output=
         Sakuli suite \\"test-suite\\" 
    
        . (Last suite run: xx)
\\\\n
        <style>
    .modalDialog {
        width: 640px;
    }

    .modalDialog:target {
        width: auto;
        margin: 20px auto;
        overflow: scroll;
        position: fixed;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        z-index: 99999;
        opacity: 1;
        pointer-events: auto;
    }

    .modalDialog:target .close {
        display: block;
    }

    .modalDialog:target .screenshot {
        width: 100%;
        border: 2px solid #333;
    }

    .screenshot {
        width: 98%;
        border: 2px solid  gray;
        display: block;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 4px;
        cursor: -webkit-zoom-in;
        cursor: -moz-zoom-in;
    }

    .close {
        display: none;
        background: #aaa;
        color: #fff;
        line-height: 25px;
        position: absolute;
        right: 10px;
        text-align: center;
        top: 25px;
        width: 65px;
        text-decoration: none;
        font-weight: bold;
        -webkit-border-radius: 12px;
        -moz-border-radius: 12px;
        border-radius: 12px;
    }

    .close:hover {
        background: #333;
    }
</style><table style=\\"border-collapse: collapse;\\"><tr valign=\\"top\\"><td class=\\"service[OK]\\">
             Sakuli suite \\"test-suite\\" 
                             
                                        . (Last suite run: 
                        xx
            )
                    </td></tr></table>
    |    suite_test-suite=0.00s;0;0;;
 graph_type_suite=0;;;;
 suite__state=;;;;
 suite__warning=0s;;;;
 suite__critical=0s;;;;
     [check_sakuli]"
`;

exports[`templates main.twig should match snapshot for case 1`] = `
"type=passive

    host_name=
            changeme
    

    start_time=1556540580.355

    finish_time=-1

    return_code=

                service_description=_test-case

        output=
         Case \\"test-case\\" of Sakuli suite \\"\\" 
    
        . (Last suite run: xx)
\\\\n
        <style>
    .modalDialog {
        width: 640px;
    }

    .modalDialog:target {
        width: auto;
        margin: 20px auto;
        overflow: scroll;
        position: fixed;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        z-index: 99999;
        opacity: 1;
        pointer-events: auto;
    }

    .modalDialog:target .close {
        display: block;
    }

    .modalDialog:target .screenshot {
        width: 100%;
        border: 2px solid #333;
    }

    .screenshot {
        width: 98%;
        border: 2px solid  gray;
        display: block;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 4px;
        cursor: -webkit-zoom-in;
        cursor: -moz-zoom-in;
    }

    .close {
        display: none;
        background: #aaa;
        color: #fff;
        line-height: 25px;
        position: absolute;
        right: 10px;
        text-align: center;
        top: 25px;
        width: 65px;
        text-decoration: none;
        font-weight: bold;
        -webkit-border-radius: 12px;
        -moz-border-radius: 12px;
        border-radius: 12px;
    }

    .close:hover {
        background: #333;
    }
</style><table style=\\"border-collapse: collapse;\\"><tr valign=\\"top\\"><td class=\\"service[OK]\\">
             Case \\"test-case\\" of Sakuli suite \\"\\" 
                             
                                                . (Last suite run: 
                        xx
            )
                    </td></tr></table>
    |         [check_sakuli]"
`;

exports[`templates main.twig should match snapshot for step 1`] = `
"type=passive

    host_name=
            changeme
    

    start_time=1556540580.754

    finish_time=-1

    return_code=

                service_description=__step-1

        output=
         Step \\"step-1\\" in case \\"\\" of Sakuli suite \\"\\" 
    
        . (Last test step run: xx)
\\\\n
        <style>
    .modalDialog {
        width: 640px;
    }

    .modalDialog:target {
        width: auto;
        margin: 20px auto;
        overflow: scroll;
        position: fixed;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        z-index: 99999;
        opacity: 1;
        pointer-events: auto;
    }

    .modalDialog:target .close {
        display: block;
    }

    .modalDialog:target .screenshot {
        width: 100%;
        border: 2px solid #333;
    }

    .screenshot {
        width: 98%;
        border: 2px solid  gray;
        display: block;
        margin-left: auto;
        margin-right: auto;
        margin-bottom: 4px;
        cursor: -webkit-zoom-in;
        cursor: -moz-zoom-in;
    }

    .close {
        display: none;
        background: #aaa;
        color: #fff;
        line-height: 25px;
        position: absolute;
        right: 10px;
        text-align: center;
        top: 25px;
        width: 65px;
        text-decoration: none;
        font-weight: bold;
        -webkit-border-radius: 12px;
        -moz-border-radius: 12px;
        border-radius: 12px;
    }

    .close:hover {
        background: #333;
    }
</style><table style=\\"border-collapse: collapse;\\"><tr valign=\\"top\\"><td class=\\"service[OK]\\">
             Step \\"step-1\\" in case \\"\\" 
                             
                                                . (Last suite run: 
                        xx
            )
                    </td></tr></table>
    |         [check_sakuli]"
`;
